File: various/CPP_VS_JAVA/simple_java/Dog.java
public class Dog {
    private String name;
    private String species;

    public Dog(String n, String s) {
        name = n;
        species = s;
    }

    public void makeSound() {
        System.out.println("Dog " + name + " of species " + species + " barks");
    }

    public static void main(String[] args) {
        Dog obj = new Dog("Adelle", "Maltese");
        obj.makeSound();
    }
}

/*
Dog Adelle of species Maltese barks
*/

File: various/CPP_VS_JAVA/simple_cpp/dog.cpp
#include <iostream>
using namespace std;
class dog
{
private:
    string name;
    string species;

public:
    dog(string n, string s) : name(n), species(s) {}

    void make_sound()
    {
        cout << "Dog " << name << " of species " << species << " barks" << endl;
    }
};

int main()
{
    dog obj("Adelle", "Maltese");
    obj.make_sound();
}

/*
Dog Adelle of species Maltese barks
*/